### PR TITLE
ENH: Extend iperf3 API with cmd line args and k8s fields

### PR DESCRIPTION
**Pull request by otto-von-bivouac**

### CODE REVIEW

File: api/v1alpha1/iperf3_types.go [hunk 1]
```diff
@@ -20,21 +20,55 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// Iperf3Spec defines the Iperf3 Benchmark Job
+// Iperf3ConfigurationSpec contains configuration parameters
+// with scheduling options for the both the iperf3 client
+// and server instances.
+type Iperf3ConfigurationSpec struct {
+	// Command line arguments appended to the
```
Comment (julio-santiesteban) [inline api/v1alpha1/iperf3_types.go]: Should start with "CmdLineArgs"

File: api/v1alpha1/iperf3_types.go [hunk 2]
```diff
@@ -20,21 +20,55 @@ import (
...
+	// predefined iperf3 parameters
+	// +optional
+	CmdLineArgs string `json:"cmdLineArgs"`
+
+	// Labels to add to the iperf3 pod.
```
Comment (julio-santiesteban) [inline api/v1alpha1/iperf3_types.go]: Should start with "PodLabels"

File: api/v1alpha1/iperf3_types.go [hunk 3]
```diff
@@ -20,21 +20,55 @@ import (
...
+	// +optional
+	PodLabels map[string]string `json:"podLabels,omitempty"`
+
+	// Scheduling related options to determine which
```
Comment (julio-santiesteban) [inline api/v1alpha1/iperf3_types.go]: Should start with...

File: api/v1alpha1/iperf3_types.go [hunk 4]
```diff
@@ -20,21 +20,55 @@ import (
...
+	// node the iperf3 pod should be scheduled
+	// +optional
+	PodScheduling PodSchedulingSpec `json:"podScheduling,omitempty"`
+
+	// Host Network requested for the iperf3 pod, if enabled the
```
Comment (julio-santiesteban) [inline api/v1alpha1/iperf3_types.go]: Should start with...

File: api/v1alpha1/iperf3_types.go [hunk 5]
```diff
@@ -20,21 +20,55 @@ import (
...
+	// Use UDP rather than TCP. If enabled the '--udp' parameter is set.
 	// +optional
 	UDP bool `json:"udp,omitempty"`
 }
 
-// Iperf3Status defines the observed state of Iperf3
+// Iperf3Status describes the current state of the benchmark
 type Iperf3Status struct {
-	// Shows if the benchmark is running
+	// State of execution
```
Comment (julio-santiesteban) [inline api/v1alpha1/iperf3_types.go]: Should start with...

File: api/v1alpha1/iperf3_types.go [hunk 6]
```diff
@@ -20,21 +20,55 @@ import (
...
 	Running bool `json:"running"`
-	// Shows completion of benchmark
+	// State of completion
```
Comment (julio-santiesteban) [inline api/v1alpha1/iperf3_types.go]: Should start with...

File: api/v1alpha1/k8s_types.go [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+// PodSchedulingSpec encapsulates the scheduling related
+// fields of a Kubernetes Pod
+type PodSchedulingSpec struct {
+	// Affinity is a group of affinity scheduling rules.
+	// +optional
+	Affinity corev1.Affinity `json:"affinity,omitempty"`
+
+	// If specified, the pod's tolerations.
```
Comment (julio-santiesteban) [inline api/v1alpha1/k8s_types.go]: Should start with...

File: api/v1alpha1/k8s_types.go [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// A node selector represents the union of the results of
```
Comment (julio-santiesteban) [inline api/v1alpha1/k8s_types.go]: Should start with...

File: api/v1alpha1/iperf3_types.go [hunk 9]
```diff
@@ -20,21 +20,55 @@ import (
...
 
-// Iperf3Spec defines the Iperf3 Benchmark Job
+// Iperf3ConfigurationSpec contains configuration parameters
+// with scheduling options for the both the iperf3 client
+// and server instances.
+type Iperf3ConfigurationSpec struct {
+	// Command line arguments appended to the
+	// predefined iperf3 parameters
+	// +optional
+	CmdLineArgs string `json:"cmdLineArgs"`
```
Comment (julio-santiesteban) [inline api/v1alpha1/iperf3_types.go]: missing omitempty?

File: api/v1alpha1/iperf3_types.go (hunk 1 above)
Comment (otto-von-bivouac) [reply api/v1alpha1/iperf3_types.go]: I also read the suggestion in vs-code quick fix, but it is not a strict requirement.
One can find examples of not repeating the variable name in the kubernetes lib too:
https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#DeleteOptions

File: api/v1alpha1/iperf3_types.go (hunk 9 above)
Comment (otto-von-bivouac) [reply api/v1alpha1/iperf3_types.go]: good point, will fix it!

File: api/v1alpha1/iperf3_types.go (hunk 1 above)
Comment (julio-santiesteban) [reply api/v1alpha1/iperf3_types.go]: https://blog.golang.org/godoc-documenting-go-code

> Notice this comment is a complete sentence that begins with the name of the element it describes. This important convention allows us to generate documentation in a variety of formats, from plain text to HTML to UNIX man pages, and makes it read better when tools truncate it for brevity, such as when they extract the first line or sentence.

File: api/v1alpha1/k8s_types.go (hunk 7 above)
Comment (otto-von-bivouac) [reply api/v1alpha1/k8s_types.go]: This is copy-pasted from kubernetes api. would leave as-is

File: api/v1alpha1/k8s_types.go (hunk 8 above)
Comment (otto-von-bivouac) [reply api/v1alpha1/k8s_types.go]: This is copy-pasted from kubernetes api. would leave as-is

**Pull request merged**